test-suite/handwritten-src/js/CppExceptionTest.js

```diff
@@ -14,9 +20,18 @@ class CppExceptionTest {
         } catch (e) {
             thrown = e;
         }
-        // In JS all we get is a number (probably a pointer)
-        // Don't know how to get to the actual exception object.
-        assertNe(thrown, null);
+        assertEq(thrown.message, "djinni: Exception Thrown");
+    }
+
+    testJsExceptionPassthrough() {
+        var thrown = null;
+        try {
+            this.cppInterface.throwJsException(new JsThrower());
+        } catch (e) {
+            thrown = e;
+        }
+        assertTrue(thrown instanceof Error);
+        assertEq(thrown.message, "JS Error");
```

Comment:
We've got tests here of catching a C++ error in JS and a JS error in JS (passing through C++).

What do you think about some tests for catching a JS error in C++ and a C++ error in C++ (passing through JS)?

Reply:
Yeah.  Also tests should check the C++ stack unwinding is performed.

Reply:
Actually,  I think we could probably do the same pass-through trick for Java code, too. In a follow-up PR.